### PR TITLE
Added __call method to message class to help suppress method not foun…

### PR DESCRIPTION
**Pull request by T3chn0crat**
Added __call method to message class to help suppress method not found errors

### CODE REVIEW

File: tests/MailThiefTest.php [hunk 1]
```diff
@@ -307,4 +307,20 @@ public function test_get_body_raw()
...
+    {
+        $mailer = $this->getMailThief();
+
+        $mailer->send('example-view', [], function ($m) {
+            $m->subject('First message')->addPart('html content', 'text/html');
+        });
+
+        $mailer->send('example-view', [], function ($m) {
+            $m->subject('Second message');
+            $m->addPart('html content', 'text/html');
```
Comment (mattstauffer) [inline tests/MailThiefTest.php]: Seems valuable to add a test that chains something *after* `addPart` so we can see that the `return $this` is making it chainable as expected, right?

**Comment by T3chn0crat** [comment]
@mattstauffer Added the test you wanted

**Comment by mattstauffer** [comment]
Hey @T3chn0crat, thanks! Sorry for the delay, I've been doing some thinking and chatting with folks about handling this well.

I realized one problem that we aren't handling here:

If someone passes an *invalid* method, MailThief will make them think that the code is OK, and they won't know it's broken until they go to run it themselves.. which is no good!

I have one quick question that I think will help us make good steps going forward--thanks for your patience with this process!

With the `addPart` method, could you explain what they're actually doing  there that can't be done with Laravel's mailer class? That might help me make a good decision.

Thanks again!!

**Comment by T3chn0crat** [comment]
@mattstauffer Yeah no problem.  I respect that you want to get it right so we shouldn't rush in the PR.

I think I said it some where else, but this isn't really ideal.  

The problem here is that there are a few methods that could be called when you compile a message that are currently not implemented with MailThief.  This cause errors in tests like #8.  

You can see https://github.com/swiftmailer/swiftmailer/blob/master/lib/classes/Swift/Message.php  and http://swiftmailer.org/docs/messages.html for more methods.

After doing a couple hours of reading and looking and testing for my response to this, I think I am on a different path than I was.

Looking at all the methods I think these are missing:
* addPart
* setReturnPath
* setReadReceiptTo
* setCharset
* setMaxLineLength
* setPriority
* attachSigner

Which isn't that many (unless I missed some).  So I guess we could do one of two things.

1. I can work on implementing these methods in MailThief
2. We can put these into an array in the __call method and if the call is one of these return `$this` if not throw an exception.

Let me know your thoughts and I will implement.

**Comment by mattstauffer** [comment]
@T3chn0crat Hm, if there are only 7, let's just keep your current PR, but like you said, add a check to that list of 7 to the `__call` method so *only* those get forwarded along.

Note that we don't need these methods:
- `priority`; Laravel has `setPriority`: https://laravel.com/api/4.2/Illuminate/Mail/Message.html#method_priority
- `returnpath`; Laravel has `setReturnpath`: https://laravel.com/api/4.2/Illuminate/Mail/Message.html#method_returnPath

**Comment by T3chn0crat** [comment]
So with out knowing it I was proving that my previous PR wasn't a good idea.  I had addParts vs addPart in my test. :/

I made those changes.  See that works for you.

**Comment by mattstauffer** [comment]
@T3chn0crat Fantastic--thanks so much! Gonna ask someone else at Tighten to review it just for my sanity, but I love this. 👍  thanks for your diilgence!

**Comment by T3chn0crat** [comment]
:+1:

**Comment by besologic** [comment]
Looks good 👍🏼

@T3chn0crat, thanks for your efforts and working with us to come up with a solid solution here. Appreciate your contributions!

**Pull request merged**